test/utils/UtilsSpec.scala

```diff
@@ -89,6 +89,28 @@ class UtilsSpec extends SpecBase {
     }
   }
 
+  "h2InnerComponent" should {
+    "create the component correctly with provided input" in new Setup {
+      h2InnerComponent(msgKey = testMsg, id = Some(testId), innerMsg = "test") mustBe
+        new h2Inner().apply(msg = testMsg, id = Some(testId), innerMsg = "test", classes = "govuk-caption-xl")
+    }
+  }
+
+  "buildCacheId" should {
+    "return a new cache id with given inputs" in {
+      buildCacheId("can1234", "$40.00") mustBe "can1234_$40.00"
+      buildCacheId("can1234", "-$20.10") mustBe "can1234_-$20.10"
+    }
+  }
+
+  "extractNumericValue" should {
+    "return only numeric value without pound or any symbols except minus" in {
```

Comment:
Could you please add one assertion for 4 digits (before decimal point) and one for 5 digits?

Also better to add an assertion for the empty string